src/lib/web-component/filter-keyword/filter-keyword-element.js

```diff
@@ -0,0 +1,39 @@
+export class FilterKeyword extends HTMLElement {
+  constructor() {
+    super();
+
+    const shadow = this.attachShadow({ mode: 'open' });
+    const filterContainer = document
+      .querySelector('#filterKeywordTemplate')
+      .content.cloneNode(true);
+
+    this.inputBox = filterContainer.querySelector('input[name="keyword"]');
+
+    shadow.appendChild(filterContainer);
+  }
+
+  connectedCallback() {
+    let timer = 0;
+    this.inputBoxListener = (event) => {
```

Comment:
The debouncing logic is quite repetitive (and has a bug - see my comment at the filter-option class).

Consider fixing the logic and putting the implementation at a shared location.

Reply:
should I use the following delay function for this?
```
  delay(fn, ms) {
    let timer = 0;
    return function (...args) {
      clearTimeout(timer);
      timer = setTimeout(fn.bind(this, ...args), ms || 0);
    };
  }
```

Reply:
https://github.com/mozilla/extension-activity-monitor/pull/14#discussion_r460060187 points to a bug. When disconnected, an element should cease to work. In the current implementation it will continue to trigger events.

A way to avoid this issue and to avoid duplication is to let the controller coalesce the repeated filter changes. It can update the filter, but debounce the filter change propagation until the value stabilizes. SInce this is an optimization that isn't required for this PR, I would not bother with it now, and defer it to later.

In the `keyword` input, there is likely some value in debouncing since a user types.
In the checkbox component, I would remove it, since there is rarely a use for debouncing checkbox changes.

So, to summarize: to keep this PR focused on what it should be doing, fix the debounce logic to clear the timer on element removal, and remove the debounce logic from the filter-option component.